doc/usecases/polyculture/add_plant_relationships.md

```diff
@@ -0,0 +1,33 @@
+# Use Case: Add Plant Relationships
+
+## Summary
+
+- **Scope:** Plants Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** A new plant relationship is added to the system.
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  - The user has the opened the UI to propose a new relationship.
+- **Main success scenario:**
+  - Details about the relationship are provided by the user.
+     - The two sides of the relationship can be
+       - Specific plants
+       - Any higher taxonomic level (e.g. family, subfamily)
```

Comment:
Which taxonomic levels are there?
Please disregard this question, if this is not within the scope of this PR.

Reply:
@Benjaminimal please define "relationship", "companion" and "antagonist" in the glossary.

Reply:
"companion" and "antagonist" are already in there.
I'll add the other term.
https://github.com/ElektraInitiative/PermaplanT/blob/c1240880cfdd47a5667c48aa0401c0a2e404b6ad/doc/architecture/glossary.md#L21-L25